File: javaclient/javaclient/models/Metrics.java
package javaclient.models;

import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import com.microsoft.kiota.serialization.UntypedNode;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
/**
 * List of basic metrics
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class Metrics implements Parsable {
    /**
     * Data used for charting etc
     */
    private UntypedNode datasets;
    /**
     * Percentage of all APIs where auto fixes have been applied
     */
    private Integer fixedPct;
    /**
     * Total number of fixes applied across all APIs
     */
    private Integer fixes;
    /**
     * Number of newly invalid APIs
     */
    private Integer invalid;
    /**
     * Open GitHub issues on our main repo
     */
    private Integer issues;
    /**
     * Number of unique APIs
     */
    private Integer numAPIs;
    /**
     * Number of methods of API retrieval
     */
    private Integer numDrivers;
    /**
     * Total number of endpoints inside all definitions
     */
    private Integer numEndpoints;
    /**
     * Number of API providers in directory
     */
    private Integer numProviders;
    /**
     * Number of API definitions including different versions of the same API
     */
    private Integer numSpecs;
    /**
     * GitHub stars for our main repo
     */
    private Integer stars;
    /**
     * Summary totals for the last 7 days
     */
    private MetricsThisWeek thisWeek;
    /**
     * Number of unofficial APIs
     */
    private Integer unofficial;
    /**
     * Number of unreachable (4XX,5XX status) APIs
     */
    private Integer unreachable;
    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param parseNode The parse node to use to read the discriminator value and create the object
     * @return a {@link Metrics}
     */
    @jakarta.annotation.Nonnull
    public static Metrics createFromDiscriminatorValue(@jakarta.annotation.Nonnull final ParseNode parseNode) {
        Objects.requireNonNull(parseNode);
        return new Metrics();
    }
    /**
     * Gets the datasets property value. Data used for charting etc
     * @return a {@link UntypedNode}
     */
    @jakarta.annotation.Nullable
    public UntypedNode getDatasets() {
        return this.datasets;
    }
    /**
     * The deserialization information for the current model
     * @return a {@link Map<String, java.util.function.Consumer<ParseNode>>}
     */
    @jakarta.annotation.Nonnull
    public Map<String, java.util.function.Consumer<ParseNode>> getFieldDeserializers() {
        final HashMap<String, java.util.function.Consumer<ParseNode>> deserializerMap = new HashMap<String, java.util.function.Consumer<ParseNode>>(14);
        deserializerMap.put("datasets", (n) -> { this.setDatasets(n.getObjectValue(UntypedNode::createFromDiscriminatorValue)); });
        deserializerMap.put("fixedPct", (n) -> { this.setFixedPct(n.getIntegerValue()); });
        deserializerMap.put("fixes", (n) -> { this.setFixes(n.getIntegerValue()); });
        deserializerMap.put("invalid", (n) -> { this.setInvalid(n.getIntegerValue()); });
        deserializerMap.put("issues", (n) -> { this.setIssues(n.getIntegerValue()); });
        deserializerMap.put("numAPIs", (n) -> { this.setNumAPIs(n.getIntegerValue()); });
        deserializerMap.put("numDrivers", (n) -> { this.setNumDrivers(n.getIntegerValue()); });
        deserializerMap.put("numEndpoints", (n) -> { this.setNumEndpoints(n.getIntegerValue()); });
        deserializerMap.put("numProviders", (n) -> { this.setNumProviders(n.getIntegerValue()); });
        deserializerMap.put("numSpecs", (n) -> { this.setNumSpecs(n.getIntegerValue()); });
        deserializerMap.put("stars", (n) -> { this.setStars(n.getIntegerValue()); });
        deserializerMap.put("thisWeek", (n) -> { this.setThisWeek(n.getObjectValue(MetricsThisWeek::createFromDiscriminatorValue)); });
        deserializerMap.put("unofficial", (n) -> { this.setUnofficial(n.getIntegerValue()); });
        deserializerMap.put("unreachable", (n) -> { this.setUnreachable(n.getIntegerValue()); });
        return deserializerMap;
    }
    /**
     * Gets the fixedPct property value. Percentage of all APIs where auto fixes have been applied
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getFixedPct() {
        return this.fixedPct;
    }
    /**
     * Gets the fixes property value. Total number of fixes applied across all APIs
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getFixes() {
        return this.fixes;
    }
    /**
     * Gets the invalid property value. Number of newly invalid APIs
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getInvalid() {
        return this.invalid;
    }
    /**
     * Gets the issues property value. Open GitHub issues on our main repo
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getIssues() {
        return this.issues;
    }
    /**
     * Gets the numAPIs property value. Number of unique APIs
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getNumAPIs() {
        return this.numAPIs;
    }
    /**
     * Gets the numDrivers property value. Number of methods of API retrieval
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getNumDrivers() {
        return this.numDrivers;
    }
    /**
     * Gets the numEndpoints property value. Total number of endpoints inside all definitions
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getNumEndpoints() {
        return this.numEndpoints;
    }
    /**
     * Gets the numProviders property value. Number of API providers in directory
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getNumProviders() {
        return this.numProviders;
    }
    /**
     * Gets the numSpecs property value. Number of API definitions including different versions of the same API
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getNumSpecs() {
        return this.numSpecs;
    }
    /**
     * Gets the stars property value. GitHub stars for our main repo
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getStars() {
        return this.stars;
    }
    /**
     * Gets the thisWeek property value. Summary totals for the last 7 days
     * @return a {@link MetricsThisWeek}
     */
    @jakarta.annotation.Nullable
    public MetricsThisWeek getThisWeek() {
        return this.thisWeek;
    }
    /**
     * Gets the unofficial property value. Number of unofficial APIs
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getUnofficial() {
        return this.unofficial;
    }
    /**
     * Gets the unreachable property value. Number of unreachable (4XX,5XX status) APIs
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getUnreachable() {
        return this.unreachable;
    }
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public void serialize(@jakarta.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        writer.writeObjectValue("datasets", this.getDatasets());
        writer.writeIntegerValue("fixedPct", this.getFixedPct());
        writer.writeIntegerValue("fixes", this.getFixes());
        writer.writeIntegerValue("invalid", this.getInvalid());
        writer.writeIntegerValue("issues", this.getIssues());
        writer.writeIntegerValue("numAPIs", this.getNumAPIs());
        writer.writeIntegerValue("numDrivers", this.getNumDrivers());
        writer.writeIntegerValue("numEndpoints", this.getNumEndpoints());
        writer.writeIntegerValue("numProviders", this.getNumProviders());
        writer.writeIntegerValue("numSpecs", this.getNumSpecs());
        writer.writeIntegerValue("stars", this.getStars());
        writer.writeObjectValue("thisWeek", this.getThisWeek());
        writer.writeIntegerValue("unofficial", this.getUnofficial());
        writer.writeIntegerValue("unreachable", this.getUnreachable());
    }
    /**
     * Sets the datasets property value. Data used for charting etc
     * @param value Value to set for the datasets property.
     */
    public void setDatasets(@jakarta.annotation.Nullable final UntypedNode value) {
        this.datasets = value;
    }
    /**
     * Sets the fixedPct property value. Percentage of all APIs where auto fixes have been applied
     * @param value Value to set for the fixedPct property.
     */
    public void setFixedPct(@jakarta.annotation.Nullable final Integer value) {
        this.fixedPct = value;
    }
    /**
     * Sets the fixes property value. Total number of fixes applied across all APIs
     * @param value Value to set for the fixes property.
     */
    public void setFixes(@jakarta.annotation.Nullable final Integer value) {
        this.fixes = value;
    }
    /**
     * Sets the invalid property value. Number of newly invalid APIs
     * @param value Value to set for the invalid property.
     */
    public void setInvalid(@jakarta.annotation.Nullable final Integer value) {
        this.invalid = value;
    }
    /**
     * Sets the issues property value. Open GitHub issues on our main repo
     * @param value Value to set for the issues property.
     */
    public void setIssues(@jakarta.annotation.Nullable final Integer value) {
        this.issues = value;
    }
    /**
     * Sets the numAPIs property value. Number of unique APIs
     * @param value Value to set for the numAPIs property.
     */
    public void setNumAPIs(@jakarta.annotation.Nullable final Integer value) {
        this.numAPIs = value;
    }
    /**
     * Sets the numDrivers property value. Number of methods of API retrieval
     * @param value Value to set for the numDrivers property.
     */
    public void setNumDrivers(@jakarta.annotation.Nullable final Integer value) {
        this.numDrivers = value;
    }
    /**
     * Sets the numEndpoints property value. Total number of endpoints inside all definitions
     * @param value Value to set for the numEndpoints property.
     */
    public void setNumEndpoints(@jakarta.annotation.Nullable final Integer value) {
        this.numEndpoints = value;
    }
    /**
     * Sets the numProviders property value. Number of API providers in directory
     * @param value Value to set for the numProviders property.
     */
    public void setNumProviders(@jakarta.annotation.Nullable final Integer value) {
        this.numProviders = value;
    }
    /**
     * Sets the numSpecs property value. Number of API definitions including different versions of the same API
     * @param value Value to set for the numSpecs property.
     */
    public void setNumSpecs(@jakarta.annotation.Nullable final Integer value) {
        this.numSpecs = value;
    }
    /**
     * Sets the stars property value. GitHub stars for our main repo
     * @param value Value to set for the stars property.
     */
    public void setStars(@jakarta.annotation.Nullable final Integer value) {
        this.stars = value;
    }
    /**
     * Sets the thisWeek property value. Summary totals for the last 7 days
     * @param value Value to set for the thisWeek property.
     */
    public void setThisWeek(@jakarta.annotation.Nullable final MetricsThisWeek value) {
        this.thisWeek = value;
    }
    /**
     * Sets the unofficial property value. Number of unofficial APIs
     * @param value Value to set for the unofficial property.
     */
    public void setUnofficial(@jakarta.annotation.Nullable final Integer value) {
        this.unofficial = value;
    }
    /**
     * Sets the unreachable property value. Number of unreachable (4XX,5XX status) APIs
     * @param value Value to set for the unreachable property.
     */
    public void setUnreachable(@jakarta.annotation.Nullable final Integer value) {
        this.unreachable = value;
    }
}

File: goclient/goclient/models/metrics.go
package models

import (
    i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91 "github.com/microsoft/kiota-abstractions-go/serialization"
)

// Metrics list of basic metrics
type Metrics struct {
    // Data used for charting etc
    datasets i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.UntypedNodeable
    // Percentage of all APIs where auto fixes have been applied
    fixedPct *int32
    // Total number of fixes applied across all APIs
    fixes *int32
    // Number of newly invalid APIs
    invalid *int32
    // Open GitHub issues on our main repo
    issues *int32
    // Number of unique APIs
    numAPIs *int32
    // Number of methods of API retrieval
    numDrivers *int32
    // Total number of endpoints inside all definitions
    numEndpoints *int32
    // Number of API providers in directory
    numProviders *int32
    // Number of API definitions including different versions of the same API
    numSpecs *int32
    // GitHub stars for our main repo
    stars *int32
    // Summary totals for the last 7 days
    thisWeek Metrics_thisWeekable
    // Number of unofficial APIs
    unofficial *int32
    // Number of unreachable (4XX,5XX status) APIs
    unreachable *int32
}
// NewMetrics instantiates a new Metrics and sets the default values.
func NewMetrics()(*Metrics) {
    m := &Metrics{
    }
    return m
}
// CreateMetricsFromDiscriminatorValue creates a new instance of the appropriate class based on discriminator value
// returns a Parsable when successful
func CreateMetricsFromDiscriminatorValue(parseNode i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable, error) {
    return NewMetrics(), nil
}
// GetDatasets gets the datasets property value. Data used for charting etc
// returns a UntypedNodeable when successful
func (m *Metrics) GetDatasets()(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.UntypedNodeable) {
    return m.datasets
}
// GetFieldDeserializers the deserialization information for the current model
// returns a map[string]func(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(error) when successful
func (m *Metrics) GetFieldDeserializers()(map[string]func(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(error)) {
    res := make(map[string]func(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(error))
    res["datasets"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetObjectValue(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.CreateUntypedNodeFromDiscriminatorValue)
        if err != nil {
            return err
        }
        if val != nil {
            m.SetDatasets(val.(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.UntypedNodeable))
        }
        return nil
    }
    res["fixedPct"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetFixedPct(val)
        }
        return nil
    }
    res["fixes"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetFixes(val)
        }
        return nil
    }
    res["invalid"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetInvalid(val)
        }
        return nil
    }
    res["issues"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetIssues(val)
        }
        return nil
    }
    res["numAPIs"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetNumAPIs(val)
        }
        return nil
    }
    res["numDrivers"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetNumDrivers(val)
        }
        return nil
    }
    res["numEndpoints"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetNumEndpoints(val)
        }
        return nil
    }
    res["numProviders"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetNumProviders(val)
        }
        return nil
    }
    res["numSpecs"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetNumSpecs(val)
        }
        return nil
    }
    res["stars"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetStars(val)
        }
        return nil
    }
    res["thisWeek"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetObjectValue(CreateMetrics_thisWeekFromDiscriminatorValue)
        if err != nil {
            return err
        }
        if val != nil {
            m.SetThisWeek(val.(Metrics_thisWeekable))
        }
        return nil
    }
    res["unofficial"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetUnofficial(val)
        }
        return nil
    }
    res["unreachable"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt32Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetUnreachable(val)
        }
        return nil
    }
    return res
}
// GetFixedPct gets the fixedPct property value. Percentage of all APIs where auto fixes have been applied
// returns a *int32 when successful
func (m *Metrics) GetFixedPct()(*int32) {
    return m.fixedPct
}
// GetFixes gets the fixes property value. Total number of fixes applied across all APIs
// returns a *int32 when successful
func (m *Metrics) GetFixes()(*int32) {
    return m.fixes
}
// GetInvalid gets the invalid property value. Number of newly invalid APIs
// returns a *int32 when successful
func (m *Metrics) GetInvalid()(*int32) {
    return m.invalid
}
// GetIssues gets the issues property value. Open GitHub issues on our main repo
// returns a *int32 when successful
func (m *Metrics) GetIssues()(*int32) {
    return m.issues
}
// GetNumAPIs gets the numAPIs property value. Number of unique APIs
// returns a *int32 when successful
func (m *Metrics) GetNumAPIs()(*int32) {
    return m.numAPIs
}
// GetNumDrivers gets the numDrivers property value. Number of methods of API retrieval
// returns a *int32 when successful
func (m *Metrics) GetNumDrivers()(*int32) {
    return m.numDrivers
}
// GetNumEndpoints gets the numEndpoints property value. Total number of endpoints inside all definitions
// returns a *int32 when successful
func (m *Metrics) GetNumEndpoints()(*int32) {
    return m.numEndpoints
}
// GetNumProviders gets the numProviders property value. Number of API providers in directory
// returns a *int32 when successful
func (m *Metrics) GetNumProviders()(*int32) {
    return m.numProviders
}
// GetNumSpecs gets the numSpecs property value. Number of API definitions including different versions of the same API
// returns a *int32 when successful
func (m *Metrics) GetNumSpecs()(*int32) {
    return m.numSpecs
}
// GetStars gets the stars property value. GitHub stars for our main repo
// returns a *int32 when successful
func (m *Metrics) GetStars()(*int32) {
    return m.stars
}
// GetThisWeek gets the thisWeek property value. Summary totals for the last 7 days
// returns a Metrics_thisWeekable when successful
func (m *Metrics) GetThisWeek()(Metrics_thisWeekable) {
    return m.thisWeek
}
// GetUnofficial gets the unofficial property value. Number of unofficial APIs
// returns a *int32 when successful
func (m *Metrics) GetUnofficial()(*int32) {
    return m.unofficial
}
// GetUnreachable gets the unreachable property value. Number of unreachable (4XX,5XX status) APIs
// returns a *int32 when successful
func (m *Metrics) GetUnreachable()(*int32) {
    return m.unreachable
}
// Serialize serializes information the current object
func (m *Metrics) Serialize(writer i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.SerializationWriter)(error) {
    {
        err := writer.WriteObjectValue("datasets", m.GetDatasets())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt32Value("fixedPct", m.GetFixedPct())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt32Value("fixes", m.GetFixes())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt32Value("invalid", m.GetInvalid())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt32Value("issues", m.GetIssues())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt32Value("numAPIs", m.GetNumAPIs())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt32Value("numDrivers", m.GetNumDrivers())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt32Value("numEndpoints", m.GetNumEndpoints())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt32Value("numProviders", m.GetNumProviders())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt32Value("numSpecs", m.GetNumSpecs())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt32Value("stars", m.GetStars())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteObjectValue("thisWeek", m.GetThisWeek())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt32Value("unofficial", m.GetUnofficial())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt32Value("unreachable", m.GetUnreachable())
        if err != nil {
            return err
        }
    }
    return nil
}
// SetDatasets sets the datasets property value. Data used for charting etc
func (m *Metrics) SetDatasets(value i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.UntypedNodeable)() {
    m.datasets = value
}
// SetFixedPct sets the fixedPct property value. Percentage of all APIs where auto fixes have been applied
func (m *Metrics) SetFixedPct(value *int32)() {
    m.fixedPct = value
}
// SetFixes sets the fixes property value. Total number of fixes applied across all APIs
func (m *Metrics) SetFixes(value *int32)() {
    m.fixes = value
}
// SetInvalid sets the invalid property value. Number of newly invalid APIs
func (m *Metrics) SetInvalid(value *int32)() {
    m.invalid = value
}
// SetIssues sets the issues property value. Open GitHub issues on our main repo
func (m *Metrics) SetIssues(value *int32)() {
    m.issues = value
}
// SetNumAPIs sets the numAPIs property value. Number of unique APIs
func (m *Metrics) SetNumAPIs(value *int32)() {
    m.numAPIs = value
}
// SetNumDrivers sets the numDrivers property value. Number of methods of API retrieval
func (m *Metrics) SetNumDrivers(value *int32)() {
    m.numDrivers = value
}
// SetNumEndpoints sets the numEndpoints property value. Total number of endpoints inside all definitions
func (m *Metrics) SetNumEndpoints(value *int32)() {
    m.numEndpoints = value
}
// SetNumProviders sets the numProviders property value. Number of API providers in directory
func (m *Metrics) SetNumProviders(value *int32)() {
    m.numProviders = value
}
// SetNumSpecs sets the numSpecs property value. Number of API definitions including different versions of the same API
func (m *Metrics) SetNumSpecs(value *int32)() {
    m.numSpecs = value
}
// SetStars sets the stars property value. GitHub stars for our main repo
func (m *Metrics) SetStars(value *int32)() {
    m.stars = value
}
// SetThisWeek sets the thisWeek property value. Summary totals for the last 7 days
func (m *Metrics) SetThisWeek(value Metrics_thisWeekable)() {
    m.thisWeek = value
}
// SetUnofficial sets the unofficial property value. Number of unofficial APIs
func (m *Metrics) SetUnofficial(value *int32)() {
    m.unofficial = value
}
// SetUnreachable sets the unreachable property value. Number of unreachable (4XX,5XX status) APIs
func (m *Metrics) SetUnreachable(value *int32)() {
    m.unreachable = value
}
type Metricsable interface {
    i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable
    GetDatasets()(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.UntypedNodeable)
    GetFixedPct()(*int32)
    GetFixes()(*int32)
    GetInvalid()(*int32)
    GetIssues()(*int32)
    GetNumAPIs()(*int32)
    GetNumDrivers()(*int32)
    GetNumEndpoints()(*int32)
    GetNumProviders()(*int32)
    GetNumSpecs()(*int32)
    GetStars()(*int32)
    GetThisWeek()(Metrics_thisWeekable)
    GetUnofficial()(*int32)
    GetUnreachable()(*int32)
    SetDatasets(value i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.UntypedNodeable)()
    SetFixedPct(value *int32)()
    SetFixes(value *int32)()
    SetInvalid(value *int32)()
    SetIssues(value *int32)()
    SetNumAPIs(value *int32)()
    SetNumDrivers(value *int32)()
    SetNumEndpoints(value *int32)()
    SetNumProviders(value *int32)()
    SetNumSpecs(value *int32)()
    SetStars(value *int32)()
    SetThisWeek(value Metrics_thisWeekable)()
    SetUnofficial(value *int32)()
    SetUnreachable(value *int32)()
}
